CMakeLists.txt

```diff
@@ -1,14 +1,59 @@
 cmake_minimum_required(VERSION 3.0)
 
+include("cmake/HunterGate.cmake")
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.14.12.tar.gz"
+    SHA1 "2d444140db0d53393b5490604006bb4f4896a444"
+)
+
+option(ENABLE_CPP11 "Build websocketpp with CPP11 features enabled." TRUE)
+
 project(websocketpp VERSION 0.7.0)
 
 set(include_install_dir "include")
 set(config_install_dir "lib/cmake/${PROJECT_NAME}")
 
+hunter_add_package(Boost COMPONENTS chrono regex system random thread)
+hunter_add_package(OpenSSL)
```

Comment:
> If the project itself doesn't uses openssl but just through boost::asio than has it to added explicit to the project?

No it should not. Only explicit dependencies needed. Other dependencies will usually be installed recursively. Usually means hunterized CMake-based projects. Since Boost use custom build scheme we need to patch it so it install OpenSSL (by option I guess?).